js/src/components/ads-account-select-control/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import AppSelectControl from '.~/components/app-select-control';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+
+/**
+ * @param {Object} props The component props
+ * @param {string} [props.value] The selected value. IF no value is defined, then the first option is selected and onChange function is triggered.
+ * @param {Function} [props.onChange] Callback when the select value changes.
+ * @return {JSX.Element} An enhanced AppSelectControl component.
+ */
+const AdsAccountSelectControl = ( {
+	value,
+	onChange = () => {},
+	...props
+} ) => {
+	const { existingAccounts: accounts = [] } = useExistingGoogleAdsAccounts();
+
+	const options = accounts.map( ( acc ) => ( {
+		value: acc.id,
+		label: `${ acc.name } (${ acc.id })`,
+	} ) );
+
+	useEffect( () => {
+		// Triggers the onChange event in order to pre-select the initial value
```

Comment:
The code comment says it "pre-select the initial value" which is not the actual behavior of this `useEffect`. Whenever the any of `options`, `onChange` or `value` is changed and the `value` is undefined, it will perform an effect.

Suggest either adjusting this code comment to align with the implementation or vice versa.

Reply:
Updated @eason9487